### PR TITLE
running unit tests via saucelabs

### PR DESCRIPTION
**Pull request by adon-at-work**
Making use of saucelabs to make sure the filters will pass unit tests in different browsers:
https://travis-ci.org/yahoo/xss-filters/jobs/74862112

The browser list:
https://github.com/yahoo/xss-filters/pull/44/files#diff-a2a3b7b0c9c3b4b93b4aebf4e3ec3cfbR10

### CODE REVIEW

**Comment by adon-at-work** [comment]
@neraliu , please review. thanks

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ node_js:
 - '0.12'
 - '0.11'
 - '0.10'
+env:
+  global:
+    - secure: iaRFUMPT3/tYf4FZhDmWScwfnRCzk8AakeqDoiNhCx/z0O6LMha8gPuJkWF1Cnl34LmCkmQa1ZeFYht50Fjst0g7SEgjC4H4GOSLTuG1bJU3fqQYIxFrODJd0Zw0RpgrwE+I1sShwm67Ywf3PAJQWsqOQAACZjZnohjbsEElpJg=
```
Comment (neraliu) [inline .travis.yml]: why r there 2 secure? how to distinguish it?

Comment (adon-at-work) [inline .travis.yml]: they are two env variables encrypted using travis key. they're provided by saucelabs, copied and pasted here.

Comment (neraliu) [inline .travis.yml]: i am curious of how do they being referenced? array index 0 and 1?

Comment (adon-at-work) [inline .travis.yml]: see: https://travis-ci.org/yahoo/xss-filters/jobs/74862112#L81
inside nodejs, referenced by `process.env.SAUCE_USERNAME`

reference: https://docs.saucelabs.com/ci-integrations/travis-ci/

File: src/xss-filters.js [hunk 2]
```diff
@@ -81,9 +81,10 @@ exports._getPrivFilters = function () {
         };
 
 
-    function getProtocol(s) {
-        s = s.split(URI_PROTOCOL_COLON, 2);
-        return (s.length === 2 && s[0]) ? s[0] : null;
+    function getProtocol(str) {
+        var s = str.split(URI_PROTOCOL_COLON, 2);
+        // str.length !== s[0].length is for older IE (e.g., v8), where delimeter residing at last will result in length equals 1, but not 2
```
Comment (neraliu) [inline src/xss-filters.js]: can the sacuelabs test for this?

Comment (adon-at-work) [inline src/xss-filters.js]: yes i think

**Comment by neraliu** [comment]
looks ok now. merged

**Pull request merged**